diff --git a/node_modules/@react-native/community-cli-plugin/dist/commands/start/index.js b/node_modules/@react-native/community-cli-plugin/dist/commands/start/index.js
index 9b457b8..ae06deb 100644
--- a/node_modules/@react-native/community-cli-plugin/dist/commands/start/index.js
+++ b/node_modules/@react-native/community-cli-plugin/dist/commands/start/index.js
@@ -94,7 +94,7 @@ const startCommand = {
         "[Deprecated] Enable plain text JavaScript log streaming for all " +
         "connected apps. This feature is deprecated and will be removed in " +
         "future.",
-      default: false,
+      default: true,
     },
   ],
 };
