diff --git a/node_modules/vision-camera-code-scanner/vision-camera-code-scanner.podspec b/node_modules/vision-camera-code-scanner/vision-camera-code-scanner.podspec
index 5807c31..f393783 100644
--- a/node_modules/vision-camera-code-scanner/vision-camera-code-scanner.podspec
+++ b/node_modules/vision-camera-code-scanner/vision-camera-code-scanner.podspec
@@ -17,4 +17,7 @@ Pod::Spec.new do |s|

   s.dependency "React-Core"
   s.dependency "GoogleMLKit/BarcodeScanning"
+  s.dependency "VisionCameraOld"
+
+
 end
